### PR TITLE
Refactor and simplify window objects

### LINKED ISSUE #235: 🚧Refactor and simplify window objects
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

The goal of this change is to simplify the `GLWindow` and `GLWindowFacade` classes by refactoring and removing code.

The windowing system has become overly complicated and harder to maintain.  The eventual addition of other platforms and the differences of how windows are managed on those other platforms will introduce more complexity.  Doing this now is ideal to not only prepare for those future changes but to simply keep the code simple.

**Notes:**
1. Remove the `GLWindowFacade` class completely.
    - 99% of the code in this class can be moved over to the `GLWindow` class for use and is all interface based from **SILK**.  This means it will be easy to test.
    - A lot of the code in this class is duplicated in `GLWindow`.
2. Create a factory class with an interface to be used to create the **SILK** window.
    - The only thing that cannot be abstracted is the single line of code tells **SILK** to create the `IWindow` object.  The idea here is to inject the factory class into the `GLWindow` class for creating the `IWindow` object.  This would make the rest of the `GlWindowFacade` code easy to migrate to `GLWindow`
3. Be careful when the `IInputContext` object is created.
    - This object needs to be created sometime in the `IWindow.Load()` method or after.  This is because that is when the **OpenGL Context** has been created and that must exist first before creating the input object.
4. It is recommended to only port over one piece of `GLWindowFacade` at a time.  Every time the code is ported over, write unit tests to make sure they behave correctly.
5. Direct interaction with the `IWindow` object created from the factory inside of the `GLWindow` class will need to be performed in each `CachedValue` property.
6. A push notification will need to be performed in the `Init()` phase in `GLWindow` to notify the rest of the system that the **OpenGL Context** has been created.
    - This needs to be done using `IReactable<GLContextData>`

### Acceptance Criteria

**This issue is finished when:**
- [x] Create new factory named `WindowFactory`
    - This will be injected into `GLWindow` for creating the **SILK** `IWindow` object
- [ ] `GLWindowFacade` removed and not used anymore
- [ ] Everything operates as it did before
- [ ] `GLWindow` contains all of the combined behavior of `GLWindow` and `GLWindowFacade`

**Port Over Events:**
- [x] Load Event
- [x] Unload Event
- [x] UpdateFrame Event
- [x] RenderFrame Event
- [x] Resize Event

**Port Over Properties:**
- [x] public Vector2 Size
- [x] public Vector2 Location
- [x] public double UpdateFrequency
- [x] public bool CursorVisible
- [x] public StateOfWindow WindowState
- [x] public VelaptorWindowBorder WindowBorder
- [x] public string Title

**Port Over Methods:**
- [x] PreInit() Method
- [x] Init() Method
- [x] Show() Method
- [x] SwapBuffers() Method
- [x] Close() Method
- [x] Dispose() Method

**Port Over Event Handlers:**
- [x] GLMouseInput_MouseScroll() Method
- [x] GLMouseMove_MouseMove() Method
- [x] GLMouseInput_MouseDown() Method
- [x] GLMouseInput_MouseUp() Method
- [x] GLKeyboardInput_KeyDown() Method
- [x] GLKeyboardInput_KeyUp() Method

**Other:**
- [x] Code documentation added if required
- [x] Unit tests added
- [x] All unit tests pass

### ToDo Items

- [x] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [X] Issue linked to the proper project
- [X] Issue linked to proper milestone

### Issue Dependencies

_No response_

### Related Work

- #209 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>

<h2 style="font-weight:bold" align="center">✅Development Checklist✅</h2>

<details open><summary>🌳Branching🌳</summary>

Does the name of the head(source) branch for this pull request have the correct **_preview feature branch_** syntax?

Syntax: _preview/feature/&lt;issue-id&gt;-&lt;description&gt;_  
Example: _preview/feature/123-my-preview-change_
- [x] Yes
- [ ] No

Does the name of the base(target) branch for this pull request have the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No

Is the head(source) branch for this pull request created from a branch with the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No
</details>

<details open><summary>🐛Bugs🐛</summary>

Contains Bug Fix(es)
- [ ] Yes
  - [ ] A bug-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>🧨Breaking Change(s)🧨</summary>

Contains [breaking change(s)](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#modifications-to-the-public-contract)
- [x] Yes
  - [x] A breaking-change-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>✨Enhancements✨</summary>

Contains enhancement(s) that add a feature or behavior.
- [ ] Yes
  - [ ] An enhancement-label label has been added to the pull request.
- [x]  No
</details>

<details open><summary>⚙️Workflow (CI/CD) Changes⚙️</summary>

Contains changes to workflow files. These changes can only be done by the project maintainers.
- [ ] Yes
  - [ ] A workflow-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>📃Documentation Updates📃</summary>

Contains changes that require updates to existing **_code_** and/or **_Velaptor product_** documentation.
- [x] Yes
  - [x] I have updated the documentation accordingly.
    - Choose the type of documentation.
    - [x] Code Documentation
      - [x] A documentation-code-label label has been added to the pull request.
    - [ ] Product Documentation
      - [ ] A documentation-product-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>🧪Unit Testing🧪</summary>

My change requires unit tests to be written.
- [x] Yes
- [ ] No
</details>

<details open><summary>🧪Manual Testing🧪</summary>

I have manually tested my changes using the project named **_VelaptorTesting_**.
- [x] Yes
- [ ] No
</details>

<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] Pull request title matches the title of the linked issue.
- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.
- [x] My code follows the coding style of this project.
  - The style is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with a coding style that should be ignored or changed, reach out, and let's discuss it.
- [x] A preview-label label has been added to the pull request.
- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vise-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - This does not mean that the tests pass in both the local and CI environment.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: Velaptor/OpenGL/GLWindow.cs [hunk 1]
```diff
@@ -11,87 +11,123 @@ namespace Velaptor.OpenGL
...
         /// <param name="systemMonitorService">Manages the systems monitors/screens.</param>
-        /// <param name="windowFacade">The internal OpenGL window facade.</param>
         /// <param name="platform">Information about the platform that is running the application.</param>
         /// <param name="taskService">Runs asynchronous tasks.</param>
         /// <param name="contentLoader">Loads various kinds of content.</param>
         /// <param name="renderer">Renders textures and primitives.</param>
+        /// <param name="keyboardInput">Holds the state of the keyboard.</param>
+        /// <param name="mouseInput">Holds the state of the mouse.</param>
+        /// <param name="glReactable">Subscribed to for push notifications.</param>
         /// <param name="glInitReactable">Provides push notifications that OpenGL has been initialized.</param>
```
Comment (kselena) [inline Velaptor/OpenGL/GLWindow.cs:81]: Grammar 
```suggestion
        /// <param name="glInitReactable">Provides a push notification that OpenGL has been initialized.</param>
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 2]
```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
...
-        /// </summary>
-        private static void GL_GLError(object? sender, GLErrorEventArgs e) => throw new Exception(e.ErrorMessage);
+        private void RunGLWindow()
+        {
+            this.glWindow.Run();
+
+            /*NOTE:
+             * Only dispose of the window here and not in the Dispose() method!!
+             *
+             * This is because this line of code will not be executed until the Window.Run() method
```
Comment (kselena) [inline Velaptor/OpenGL/GLWindow.cs:298]: Grammar
```suggestion
             * This is because that line of code will not be executed until the Window.Run() method
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 3]
```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
...
+             * has finished executing.  This happens once the window is closed.
+             *
+             * If you dispose of the window in the Dispose() method before the Run() method is finished
```
Comment (kselena) [inline Velaptor/OpenGL/GLWindow.cs:301]: Sentence structure - for this line and the next line
```suggestion
             * If you dispose of the window in the Dispose() method before the Run() method is finished, the application will crash.
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 4]
```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
...
+            this.glInputContext.Mice[0].Scroll += GLMouseInput_MouseScroll;
+        }
 
         /// <summary>
         /// Invokes the <see cref="Initialize"/> action property.
         /// </summary>
-        private void GameWindow_Load(object? sender, EventArgs e)
+        private void GLWindow_Load()
         {
             // OpenGL is ready to take function calls after this Init() call has ran
```
Comment (kselena) [inline Velaptor/OpenGL/GLWindow.cs:355]: Grammar
```suggestion
            // OpenGL is ready to take function calls after this Init() call has run
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 5]
```diff
@@ -476,51 +632,108 @@ float ToMonitorScale(float value)
 
             if (mainMonitor is not null)
             {
+                // Set the default position to be in the center of the monitor
```
Comment (kselena) [inline Velaptor/OpenGL/GLWindow.cs:635]: Grammar
```suggestion
                // Sets the default position to be in the center of the monitor
```

**Review by kselena — CHANGES_REQUESTED** [review]

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 6]
```diff
@@ -516,6 +698,36 @@ public void MouseCursorVisible_WhenSettingValueAndNotCaching_ReturnsCorrectResul
...
+            {
+                _ = window.WindowState;
+            }, "The enum 'Silk.NET.Windowing.WindowState' is invalid because it is out of range.");
+        }
+
+        [Fact]
+        public void WindowState_WhenSettingInvalidValue_ThrowsException()
+        {
+            // Arrange
+            // this.mockSilkWindow.SetupSet(p => p.WindowState).Returns((SilkWindowBorder)1234);
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:720]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 7]
```diff
@@ -602,26 +848,76 @@ public void TypeOfBorder_WhenCachingValue_ReturnsCorrectResult()
...
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<NoMouseException>(() =>
+            {
+                window.Show();
+            }, "Input Exception: No connected mice available.");
+        }
+
         [Fact]
         public void Show_WhenInvoked_InitializesWindowFacade()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:922]: Remove this unit test named `Show_WhenInvoked_InitializesWindowFacade`

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 8]
```diff
@@ -646,7 +942,7 @@ public void Show_WhenInvoked_RunsWindowFacade()
             window.Show();
 
             // Assert
-            this.mockWindowFacade.Verify(m => m.Show(), Times.Once());
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:649]: Remove the unit test named `Show_WhenInvoked_RunsWindowFacade`

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 9]
```diff
@@ -656,7 +952,7 @@ public void Show_WhenInvoked_SetsUpOpenGLErrorCallback()
             var window = CreateWindow();
             window.Show();
 
-            this.mockWindowFacade.Raise(i => i.Load += null, EventArgs.Empty);
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:659]: Verify that the error callback is being setup properly in the unit test `Show_WhenInvoked_SetsUpOpenGLErrorCallback`

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 10]
```diff
@@ -681,8 +977,8 @@ public void Show_WhenUpdatingFrameWhileShuttingDown_DoesNotInvokeUpdateEvent()
 
             // Act
             window.Show();
-            this.mockWindowFacade.Raise(m => m.Unload += null, EventArgs.Empty);
-            this.mockWindowFacade.Raise(m => m.UpdateFrame += null, new FrameTimeEventArgs(123));
+            // this.mockWindowFacade.Raise(m => m.Unload += null, EventArgs.Empty);
+            // this.mockWindowFacade.Raise(m => m.UpdateFrame += null, new FrameTimeEventArgs(123));
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:981]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 11]
```diff
@@ -702,8 +998,8 @@ public void Show_WhenRenderingFrameWhileShuttingDown_DoesNotInvokeRenderEvent()
 
             // Act
             window.Show();
-            this.mockWindowFacade.Raise(m => m.Unload += null, EventArgs.Empty);
-            this.mockWindowFacade.Raise(m => m.RenderFrame += null, new FrameTimeEventArgs(234));
+            // this.mockWindowFacade.Raise(m => m.Unload += null, EventArgs.Empty);
+            // this.mockWindowFacade.Raise(m => m.RenderFrame += null, new FrameTimeEventArgs(234));
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1002]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 12]
```diff
@@ -726,11 +1022,11 @@ public async void ShowAsync_WhenInvoked_SubscribesToWindowEvents()
...
+            // this.mockWindowFacade.VerifyAdd(s => s.Load += It.IsAny<EventHandler<EventArgs>>(), Times.Once(), $"Subscription of the '{nameof(IGameWindowFacade.Load)}' event did not occur.");
+            // this.mockWindowFacade.VerifyAdd(s => s.Unload += It.IsAny<EventHandler<EventArgs>>(), Times.Once(), $"Subscription of the '{nameof(IGameWindowFacade.Unload)}' event did not occur.");
+            // this.mockWindowFacade.VerifyAdd(s => s.UpdateFrame += It.IsAny<EventHandler<FrameTimeEventArgs>>(), Times.Once(), $"Subscription of the '{nameof(IGameWindowFacade.UpdateFrame)}' event did not occur.");
+            // this.mockWindowFacade.VerifyAdd(s => s.RenderFrame += It.IsAny<EventHandler<FrameTimeEventArgs>>(), Times.Once(), $"Subscription of the '{nameof(IGameWindowFacade.RenderFrame)}' event did not occur.");
+            // this.mockWindowFacade.VerifyAdd(s => s.Resize += It.IsAny<EventHandler<WindowSizeEventArgs>>(), Times.Once(), $"Subscription of the '{nameof(IGameWindowFacade.Resize)}' event did not occur.");
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1029]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 13]
```diff
@@ -747,10 +1043,10 @@ public async void ShowAsync_WhenInvoked_InitWindowFacade()
 
             // Act
             await window.ShowAsync();
-            this.mockWindowFacade.Raise(w => w.Load += null, EventArgs.Empty);
+            // this.mockWindowFacade.Raise(w => w.Load += null, EventArgs.Empty);
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1046]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 14]
```diff
@@ -747,10 +1043,10 @@ public async void ShowAsync_WhenInvoked_InitWindowFacade()
...
 
             // Assert
-            this.mockWindowFacade.Verify(m => m.Init(It.IsAny<uint>(), It.IsAny<uint>()), Times.Once());
+            // this.mockWindowFacade.Verify(m => m.Init(It.IsAny<uint>(), It.IsAny<uint>()), Times.Once());
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1049]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 15]
```diff
@@ -769,7 +1065,7 @@ public async void ShowAsync_WhenInvoked_ExecutesWindowShow()
             await window.ShowAsync();
 
             // Assert
-            this.mockWindowFacade.Verify(m => m.Show(), Times.Once());
+            // this.mockWindowFacade.Verify(m => m.Show(), Times.Once());
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1068]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 16]
```diff
@@ -825,13 +1166,59 @@ public void GLWindow_WhenWindowResizes_SetsGLViewportAndTriggersResizeEvent()
...
+            this.mockSilkWindow.Raise(e => e.Update += It.IsAny<Action<double>>(), 0.016);
+
+            // Assert
+            Assert.False(windowUpdateInvoked, $"{nameof(GLWindow.Update)} should not of been invoked during window shutdown.");
+            this.mockMouseInput.VerifyNever(m => m.SetScrollWheelSpeed(It.IsAny<int>()));
+            this.mockMouseInput.VerifyNever(m => m.SetScrollWheelDirection(It.IsAny<MouseScrollDirection>()));
+        }
+
+        [Fact]
+        public void GLWindow_WhenUpdatingWhenNotShuttingDown_PerformsUpdate()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1198]: Refactor unit test name
```suggestion
        public void GLWindow_WhileUpdatingWhenNotShuttingDown_PerformsUpdate()
```

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 17]
```diff
@@ -857,7 +1244,7 @@ public void GLWindow_WhenRenderingFrameWithAutoClearDisabled_ClearsGLBuffer()
             window.Show();
 
             // Act
-            this.mockWindowFacade.Raise(e => e.RenderFrame += null, null, new FrameTimeEventArgs(16));
+            // this.mockWindowFacade.Raise(e => e.RenderFrame += null, null, new FrameTimeEventArgs(16));
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1247]: Remove commented code

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 18]
```diff
@@ -872,45 +1259,177 @@ public void GLWindow_WhenRenderingFrame_InvokesDrawAndSwapsBuffer()
...
+            this.mockSilkMouse.Raise(e => e.MouseDown += It.IsAny<Action<IMouse, SilkMouseButton>>(),
+                null,
+                SilkMouseButton.Left);
+
+            // Assert
+            this.mockMouseInput.VerifyOnce(m => m.SetState(VelaptorMouseButton.LeftButton, true));
+        }
+
+        [Fact]
+        public void GLWindow_WhenMouseButtonIsReleasedDown_UpdatesMouseInputState()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:1367]: Refactor unit test name

```suggestion
        public void GLWindow_WhenMouseButtonIsReleased_UpdatesMouseInputState()
```

File: Velaptor/Factories/IInputFactory.cs [hunk 19]
```diff
@@ -0,0 +1,20 @@
+﻿// <copyright file="IInputFactory.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Velaptor.Factories
+{
+    using Silk.NET.Input;
+
+    /// <summary>
+    /// Creates an input object that gives access to native system input hardware.
```
Comment (CalvinWilkinson) [inline Velaptor/Factories/IInputFactory.cs:10]: Refactor comment

```suggestion
    /// Creates an input object that gives access to native input hardware.
```

File: Velaptor/Factories/IInputFactory.cs [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+    /// </summary>
+    internal interface IInputFactory
+    {
+        /// <summary>
+        /// Creates a new input object.
+        /// </summary>
+        /// <returns>The input object to gain access to the system input.</returns>
```
Comment (CalvinWilkinson) [inline Velaptor/Factories/IInputFactory.cs:17]: Refactor comment

```suggestion
        /// <returns>The input object to gain access to the native input hardware.</returns>
```

File: Velaptor/Factories/IWindowFactory.cs [hunk 21]
```diff
@@ -0,0 +1,20 @@
+﻿// <copyright file="IWindowFactory.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Velaptor.Factories
+{
+    using SILKWindow = Silk.NET.Windowing.IWindow;
+
+    /// <summary>
+    /// Creates different native windows.
```
Comment (CalvinWilkinson) [inline Velaptor/Factories/IWindowFactory.cs:10]: Refactor comment

```suggestion
    /// Creates a native window.
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 22]
```diff
@@ -11,87 +11,123 @@ namespace Velaptor.OpenGL
...
+    using VelaptorIWindow = Velaptor.UI.IWindow;
+    using VelaptorMouseButton = Velaptor.Input.MouseButton;
+    using VelaptorWindow = Velaptor.UI.Window;
+    using VelaptorWindowBorder = Velaptor.WindowBorder;
 
     // ReSharper restore RedundantNameQualifier
 
     /// <summary>
-    /// An OpenGL window implementation to be used inside of the <see cref="Window"/> class.
+    /// An OpenGL window implementation to be used inside of the <see cref="VelaptorWindow"/> class.
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:40]: Fully qualify the <see/> reference to the velaptor window

File: Velaptor/OpenGL/GLWindow.cs [hunk 23]
```diff
@@ -11,87 +11,123 @@ namespace Velaptor.OpenGL
...
         /// <param name="glfwInvoker">Invokes GLFW functions.</param>
         /// <param name="systemMonitorService">Manages the systems monitors/screens.</param>
-        /// <param name="windowFacade">The internal OpenGL window facade.</param>
         /// <param name="platform">Information about the platform that is running the application.</param>
         /// <param name="taskService">Runs asynchronous tasks.</param>
         /// <param name="contentLoader">Loads various kinds of content.</param>
         /// <param name="renderer">Renders textures and primitives.</param>
+        /// <param name="keyboardInput">Holds the state of the keyboard.</param>
+        /// <param name="mouseInput">Holds the state of the mouse.</param>
+        /// <param name="glReactable">Subscribed to for push notifications.</param>
```
[the remark this answers is not in the document]
Comment (CalvinWilkinson) [reply Velaptor/OpenGL/GLWindow.cs:80]: Change to below:
```suggestion
        /// <param name="glReactable">Subscribed to for OpenGL related push notifications.</param>
```

Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:80]: Refactor param name to `glContextReactable`

File: Velaptor/OpenGL/GLWindow.cs (hunk 2 above)
Comment (CalvinWilkinson) [reply Velaptor/OpenGL/GLWindow.cs:298]: Refactor to below instead

```suggestion
             * This is because the line of code below will not be executed until the Window.Run() method
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 24]
```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
...
+             * This is because this line of code will not be executed until the Window.Run() method
+             * has finished executing.  This happens once the window is closed.
+             *
+             * If you dispose of the window in the Dispose() method before the Run() method is finished
+             * then the application will crash.
+             */
+            this.glWindow.Dispose();
+        }
+
+        private void PreInit()
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:307]: Add code docs to the `PreInit()` method.

File: Velaptor/OpenGL/GLWindow.cs [hunk 25]
```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
...
+
+            this.glWindow.UpdatesPerSecond = 120;
+            this.glWindow.Load += GLWindow_Load;
+            this.glWindow.Closing += GLWindow_Closing;
+            this.glWindow.Resize += GLWindow_Resize;
+            this.glWindow.Update += GLWindow_Update;
+            this.glWindow.Render += GLWindow_Render;
+        }
+
+        private void Init(uint width, uint height)
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:324]: Add code docs to the `Init()` method.

File: Velaptor/OpenGL/GLWindow.cs (hunk 4 above)
Comment (CalvinWilkinson) [reply Velaptor/OpenGL/GLWindow.cs:355]: Change to the suggestion below:

```suggestion
            // OpenGL is ready to take function calls after this Init() call has executed
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 26]
```diff
@@ -369,21 +465,74 @@ private void GameWindow_RenderFrame(object? sender, FrameTimeEventArgs e)
 
             Draw?.Invoke(frameTime);
 
-            this.windowFacade.SwapBuffers();
+            this.glWindow.SwapBuffers();
         }
 
         /// <summary>
-        /// Sets up the OpenGL window.
+        /// Invoked when any keyboardInput key transitions from the up position to the down position.
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:472]: Refactor to below

```suggestion
        /// Invoked when any keyboard input key transitions from the up position to the down position.
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 27]
```diff
@@ -369,21 +465,74 @@ private void GameWindow_RenderFrame(object? sender, FrameTimeEventArgs e)
...
+        /// </summary>
+        /// <param name="keyboard">The system keyboardInput.</param>
+        /// <param name="key">The key that was pushed down.</param>
+        /// <param name="arg3">Additional argument from OpenGL.</param>
+        private void GLKeyboardInput_KeyDown(IKeyboard keyboard, Key key, int arg3) => this.keyboardInput.SetState((KeyCode)key, true);
+
+        /// <summary>
+        /// Invoked when any keyboardInput key transitions from the down position to the up position.
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:480]: Refactor to below

```suggestion
        /// Invoked when any keyboard input key transitions from the down position to the up position.
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 28]
```diff
@@ -369,21 +465,74 @@ private void GameWindow_RenderFrame(object? sender, FrameTimeEventArgs e)
...
+            };
+
+            this.mouseInput.SetScrollWheelDirection(wheelDirection);
+        }
+
+        /// <summary>
+        /// Invoked when the mouse moves over the window.
+        /// </summary>
+        /// <param name="mouse">The system mouse object.</param>
+        /// <param name="position">The position of the mouseInput.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:524]: Refactor comment to below

```suggestion
        /// <param name="position">The position of the mouse.</param>
```

**Review by CalvinWilkinson — COMMENTED** [review]
Review complete

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 29]
```diff
@@ -31,35 +43,62 @@ public class GLWindowTests
...
         /// </summary>
         public GLWindowTests()
         {
+            this.mockGLContext = new Mock<IGLContext>();
+            this.mockSilkWindow = new Mock<IWindow>();
+            this.mockSilkWindow.SetupGet(p => p.GLContext).Returns(this.mockGLContext.Object);
+            this.mockSilkWindow.Setup(m => m.Run(It.IsAny<Action>()))
+                .Callback<Action>(_ =>
+                {
+                    // Mock the behavior of the load event being invoked when the app is ran
```
[the remark this answers is not in the document]
Comment (CalvinWilkinson) [reply Testing/VelaptorTests/OpenGL/GLWindowTests.cs]: I changed it to "executed" instead of "runs"
